data/transactions/logic/eval.go

```diff
@@ -264,19 +259,121 @@ type EvalParams struct {
 	// MinTealVersion is nil, we will compute it ourselves
 	MinTealVersion *uint64
```

Comment:
It's because existing code uses the nil as a flag to determine the min version internally.  Code that evaluates logicsigs makes an EvalParams by hand (which leaves the value nil) and then inside the evaluator, if it sees a nil, it fills it in.

I started to change it, but we invoke it that way from regular logicsig processing, dry run, goal, debugger... it just seemed like too much to change.